### PR TITLE
Merge `main` at `3bafbea` into `next`

### PR DESCRIPTION
**Pull request by sophschneider**
Merge `main` at 3bafbea5000207c05d2c58dc3553790164ab7844 into `next`
Following this process https://github.com/Shopify/polaris/issues/10594

## Major merge changes
- [x] Merge https://github.com/Shopify/polaris/commit/3bafbea5000207c05d2c58dc3553790164ab7844 and reconcile consolidated styles @mattkubej 
- [x] Merge https://github.com/Shopify/polaris/commit/3efbc1b4ec637f43c4cb97d90830b14e486a1d22#diff-7692b6b9f0ed6aaec69d853b98131666ed9bf4db1362aeea6c60eefe83cd4d03 and reconcile consolidated styles @mrcthms 
- [x] Merge https://github.com/Shopify/polaris/commit/b53189c727bd62d67b641ae906b3af1a5bf60de7 and reconcile migrated API changes from next @kyledurand 
- [x] New shadow tokens @lgriffee @aaronccasanova

### CODE REVIEW

File: polaris-react/src/components/ActionMenu/components/SecondaryAction/SecondaryAction.scss [hunk 1]
```diff
@@ -59,12 +57,11 @@
       }
 
       // stylelint-disable-next-line selector-max-specificity -- apply hover styles
-      &:hover {
+      &:is(:hover, :focus) {
```
Comment (sophschneider) [inline polaris-react/src/components/ActionMenu/components/SecondaryAction/SecondaryAction.scss:60]: @mrcthms There weren't any `&.destructive` styles in our feature branch but I applied your changes to this `&.critical` class, hopefully that works!

Comment (mrcthms) [reply polaris-react/src/components/ActionMenu/components/SecondaryAction/SecondaryAction.scss:60]: Sounds fine to me!

**Review by mrcthms — APPROVED** [review]
Approved changes to SecondaryAction / Page Header

**Review by aaronccasanova — APPROVED** [review]
Polaris tokens updates look good 👍

**Comment by sophschneider** [comment]
Going to merge this, reviewers lmk if there are any regressions and we can do a follow up fix

**Pull request merged**